kotlin/internal/jvm/plugins.bzl

```diff
@@ -19,40 +19,51 @@ load(
 KtJvmPluginInfo = provider(
     doc = "This provider contains the plugin info for the JVM aspect",
     fields = {
-        "annotation_processors": "a serializeable list of structs containing annotation processor definitions",
+        "annotation_processors": "depset of structs containing annotation processor definitions",
     },
 )
 
-_EMPTY_PLUGIN_INFO = [KtJvmPluginInfo(annotation_processors = [])]
+_EMPTY_PLUGIN_INFO = [KtJvmPluginInfo(annotation_processors = depset())]
 
+# Mapping functions for args.add_all.
+# These preserve the transitive depsets until needed.
```

Comment:
Wait... they do? I thought `[x for x in y]` was a list comprehension. This doesn't convert `y` into a list?

Reply:
If you use args.add_all(<depset<Foo>>, map_each=fooToString) the depset isn't flattened until the command line is needed. 

However, this looks like a partial change. I think I merged it wrong. They should be processor.class and processor.cp

I'll go look.

Reply:
A commit snuck on the wrong pr. Resolve.